main.go

```diff
@@ -280,54 +278,70 @@ func (s *Session) firstNav(ctx context.Context) error {
 
-	// try jumping to the end of the page. detect we are there and have stopped
-	// moving when two consecutive screenshots are identical.
-	var previousScr, scr []byte
+	// Try scrolling to the end of the page.
+	// After each scroll attempt we extract the last Photo Page Element (href) from the DOM.
+	// We detect we are at the end when two consecutive DOM extractions are identical.
+	var previousHref string
 	for {
 		chromedp.KeyEvent(kb.PageDown).Do(ctx)
 		chromedp.KeyEvent(kb.End).Do(ctx)
-		chromedp.CaptureScreenshot(&scr).Do(ctx)
-		if previousScr == nil {
-			previousScr = scr
-			continue
+		time.Sleep(tick) // sleep *before* we establish new state in DOM
+
+		// Extract last Photo Page Element (href) from DOM.
+		lastHrefInDOM, err := lastPhotoInDOM(ctx)
+		if err != nil {
+			continue // Just ignore this error, continue will retry.
 		}
-		if bytes.Equal(previousScr, scr) {
+		if previousHref == lastHrefInDOM {
 			break
 		}
-		previousScr = scr
-		time.Sleep(tick)
+		previousHref = lastHrefInDOM
 	}
 
-	if *verboseFlag {
-		log.Printf("Successfully jumped to the end")
+	// Now that we have stopped scrolling, select (focus) on the last element
+	// The element must be focused, so that navToLast can send "\n" to enter photo detail page
+	lastEltSel := fmt.Sprintf(`a[href="%s"]`, previousHref)
+	if err := chromedp.Focus(lastEltSel).Do(ctx); err != nil {
+		log.Printf("Error focus: %s", lastEltSel)
 	}
 
+	if *verboseFlag {
+		log.Printf("Successfully jumped to the end: %s", previousHref)
+	}
 	return nil
 }
 
+// When in the Main/Album Page, the DOM contains <a href=".." /> elements for all visible images.
+// lastPhotoInDOM simply returns the last such href in document order.
+// The DOM actually contains more images than those that are visible, in a kind of virtual scrolling window
+// In the DOM, but not reflecting exactly the visible photos (actually a superset of the visible elements):
+//  <a href="./photo/AF1QipAAAAAA" aria=label="Photo - Portrait - Jul 15, 2010, 2:10:48 PM"/>
+//  <a href="./photo/AF1QipBBBBBB" aria-label="Photo - Landscape - Jul 15, 2010, 2:03:10 PM"/>
+//  <a href="./photo/AF1QipCCCCCC" aria-label="Video - Landscape - Jul 30, 2010, 7:20:22 PM"/>
+// We tried to find the actual *oldest* photo by using the aria-label attribute which contains a date for the photo,
+// unfortunately that label is localised for each user's language which makes the date format very hard to parse.
+func lastPhotoInDOM(ctx context.Context) (string, error) {
+	sel := `a[href^="./photo/"]` // css selector for all links to images with href prefix "./photo/..."
+	var attrs []map[string]string
+	if err := chromedp.AttributesAll(sel, &attrs).Do(ctx); err != nil {
+		log.Printf("lastPhotoInDOM: document.quertSelectorAll:%s error %s", sel, err)
+		return "", err
+	}
+	if len(attrs) == 0 {
+		return "", fmt.Errorf("lastPhotoInDOM: no elements match")
+	}
+
+	lastElement := attrs[len(attrs)-1]
+	href := lastElement["href"]
```

Comment:
Is this guaranteed to exist, or can we end up with the empty string here?

Reply:
Yes, It is guarateed to exist, 
- first because `len(attrs)>0` (or we would have returned an error, (which is ignored for retry in the caller), 
- and the DOM Selector itself (`a[href^="./photo/"]`) separately guarantees that any returned `href` will start with "./photo/...", *(that is the meaning of the `^=` operator in the css selector)*